lib/glific/third_party/sheets/sheets.ex

```
@@ -295,6 +296,10 @@ defmodule Glific.Sheets do
     %SheetData{}
     |> SheetData.changeset(attrs)
     |> Repo.insert()
+    |> case do
+      {:ok, _sheet_data} -> Glific.Metrics.increment("Sheets Sync Successful")
+      {:error, _changeset} -> Glific.Metrics.increment("Sheets Sync Failed")
```

Here we are inserting each row of the sheet, and this could fail if a sheet has same `Key` column since its a unique constraint and is expected to fail silently. So that doesn't mean the entire sheet sync is failed.

---

Good catch, I figured this out only after pushing this. Making the change now